rcon/player_history.py

```diff
@@ -217,13 +232,17 @@ def ban_if_blacklisted(rcon, steam_id_64, name):
             return
 
         if player.blacklist and player.blacklist.is_blacklisted:
-            logger.warning("Player %s was banned due blacklist, reason: %s", str(
+            logger.warning("Player %s is blacklisted and has been banned, reason: %s", str(
                 player), player.blacklist.reason)
-            rcon.do_perma_ban(name, player.blacklist.reason)
             # TODO save author of blacklist
             safe_save_player_action(
-                rcon=rcon, player_name=player, action_type="PERMABAN", reason=player.blacklist.reason, by='BLACKLIST'
+                rcon=rcon, player_name=name, action_type="PERMABAN", reason=player.blacklist.reason, by='BLACKLIST'
             )
+            try:
+                ban_and_add_log(rcon, steam_id_64, name, player.blacklist.reason)
+                sess.commit()
+            except:
+                logger.warning("Something went wrong banning player %s", str(player))
```

Comment:
`player.to_dict()` for be more useful to log, cause I think you'll just get the object id here otherwise